### PR TITLE
NGAlert: Consolidate on standard errors package

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
In general, Grafana uses the standard [errors](https://golang.org/pkg/errors/) package, but the ngalert package uses the non-standard [github.com/pkg/errors](https://github.com/pkg/errors) alternative instead. That we have two different error libraries throughout the codebase causes confusion and also another dependency, that could have been avoided. Bugs could potentially also arise from this, in case someone confusedly depends on details of errors made by the github.com/pkg/errors library.

Based on a previous conversation in Slack around this, we should IMHO consolidate on the standard errors package since it's A, standard, B, what we already use throughout the majority of the codebase, and C, because github.com/pkg/errors is in maintenance mode (after the standard errors package was introduced). If we were to consolidate on the github.com/pkg/errors library instead, there would have to be some weighty arguments in favour of it in light of the drawbacks previously listed.

### CODE REVIEW

**Review by codesome — APPROVED** [review]
Thanks!

File: pkg/services/ngalert/notifier/alertmanager.go [hunk 1]
```diff
@@ -136,7 +137,7 @@ func (am *Alertmanager) Init() (err error) {
 func (am *Alertmanager) Run(ctx context.Context) error {
 	// Make sure dispatcher starts. We can tolerate future reload failures.
 	if err := am.SyncAndApplyConfigFromDatabase(); err != nil {
-		am.logger.Error(errors.Wrap(err, "unable to sync configuration").Error())
+		am.logger.Error(fmt.Sprintf("unable to sync configuration: %s", err))
```
Comment (wbrowne) [inline pkg/services/ngalert/notifier/alertmanager.go:140]: Would this not be more appropriate here?

```suggestion
		am.logger.Error("unable to sync configuration", "err", err)
```

Comment (aknuds1) [reply pkg/services/ngalert/notifier/alertmanager.go:140]: @wbrowne oh yeah, good point.

File: pkg/services/ngalert/notifier/alertmanager.go [hunk 2]
```diff
@@ -146,7 +147,7 @@ func (am *Alertmanager) Run(ctx context.Context) error {
 			return nil
 		case <-time.After(pollInterval):
 			if err := am.SyncAndApplyConfigFromDatabase(); err != nil {
-				am.logger.Error(errors.Wrap(err, "unable to sync configuration").Error())
+				am.logger.Error(fmt.Sprintf("unable to sync configuration: %s", err))
```
Comment (wbrowne) [inline pkg/services/ngalert/notifier/alertmanager.go:150]: Last one I think..

```suggestion
				am.logger.Error("unable to sync configuration", "err", err)
```

**Review by wbrowne — APPROVED** [review]

**Comment by marefr** [comment]
What's our view on internal [errutil package](https://github.com/grafana/grafana/blob/561fd4162ea360264ed263b96bbbcac86ac44814/pkg/util/errutil/errors.go#L8)? Feels like using that could clean up the code a bit, but no biggie really.

**Comment by aknuds1** [comment]
> What's our view on internal [errutil package](https://github.com/grafana/grafana/blob/561fd4162ea360264ed263b96bbbcac86ac44814/pkg/util/errutil/errors.go#L8)? Feels like using that could clean up the code a bit, but no biggie really.

@marefr I think we should stop using errutil.Wrap in favour of `fmt.Errorf`, I recall @papagian voicing the same opinion and that we should do a cleanup at some point.

**Comment by marefr** [comment]
Yeah makes sense. But it's very easy to format an error wrongly and our internal package helps developers do the right thing. Something I would have hoped the standard package should have helped with, but unfortunately not.

**Comment by aknuds1** [comment]
Personally I don't have a problem with fmt.Errorf, it's used all over so might as well get used to it.

**Comment by aknuds1** [comment]
It's worth mentioning BTW that someone (on the Go team?) made the point that you don't always want to wrap errors, since you might not want to make internal errors part of your API. So it's worth thinking over before you wrap.

**Comment by papagian** [comment]
I don't remember a lot but I think that we had this discussion about `errutil.Wrap` before changing it to use the standard library instead of `golang.org/x/xerrors`.
I don't have a strong opinion but since it finally calls the standard library I think it's fine to keep it for convenience if someone wants to use it.

**Review by papagian — APPROVED** [review]
LGTM ~but it would be good to wait for @gotjosh's final~ 👌  @codesome has approved also so I think it's fine!

Fixes https://github.com/grafana/alerting-squad/issues/75

**Pull request merged**